src/snowflake/cli/plugins/notebook/manager.py

```diff
@@ -1,16 +1,48 @@
+from pathlib import Path
+
+from snowflake.cli.api.cli_global_context import cli_context
 from snowflake.cli.api.identifiers import FQN
 from snowflake.cli.api.sql_execution import SqlExecutionMixin
 from snowflake.cli.plugins.connection.util import make_snowsight_url
 
+from .exceptions import NotebookStagePathError
+from .types import NotebookName, NotebookStagePath
+
 
 class NotebookManager(SqlExecutionMixin):
-    def execute(self, notebook_name: str):
+    def execute(self, notebook_name: NotebookName):
         query = f"EXECUTE NOTEBOOK {notebook_name}()"
         return self._execute_query(query=query)
 
-    def get_url(self, notebook_name: str):
+    def get_url(self, notebook_name: NotebookName):
         fqn = FQN.from_string(notebook_name).using_connection(self._conn)
         return make_snowsight_url(
             self._conn,
             f"/#/notebooks/{fqn.url_identifier}",
         )
+
+    @staticmethod
+    def _parse_stage_path(notebook_file: NotebookName) -> Path:
+        if not notebook_file.endswith(".ipynb"):
+            raise NotebookStagePathError(notebook_file)
+        stage_path = Path(notebook_file)
+        if len(stage_path.parts) < 2:
+            raise NotebookStagePathError(notebook_file)
+
+        return stage_path
+
+    def create(self, notebook_name: NotebookName, notebook_file: NotebookStagePath):
```

Comment:
Please change `notebook_name` to FQN. Current implementation will cause problems with quoted identifiers.

Reply:
🫡